internal/handlers/sqlite/msg_aggregate.go

```diff
@@ -16,13 +16,335 @@ package sqlite
 
 import (
 	"context"
+	"errors"
+	"fmt"
+	"math"
+	"time"
 
+	"github.com/FerretDB/FerretDB/internal/backends"
+	"github.com/FerretDB/FerretDB/internal/clientconn/conninfo"
+	"github.com/FerretDB/FerretDB/internal/clientconn/cursor"
+	"github.com/FerretDB/FerretDB/internal/handlers/common"
+	"github.com/FerretDB/FerretDB/internal/handlers/common/aggregations"
+	"github.com/FerretDB/FerretDB/internal/handlers/common/aggregations/stages"
+	"github.com/FerretDB/FerretDB/internal/handlers/commonerrors"
+	"github.com/FerretDB/FerretDB/internal/handlers/commonparams"
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/iterator"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
 	"github.com/FerretDB/FerretDB/internal/util/must"
 	"github.com/FerretDB/FerretDB/internal/wire"
 )
 
 // MsgAggregate implements HandlerInterface.
 func (h *Handler) MsgAggregate(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, error) {
-	// TODO https://github.com/FerretDB/FerretDB/issues/3148
-	return nil, notImplemented(must.NotFail(msg.Document()).Command())
+	document, err := msg.Document()
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	common.Ignored(document, h.L, "lsid")
+
+	if err = common.Unimplemented(document, "explain", "collation", "let"); err != nil {
+		return nil, err
+	}
+
+	common.Ignored(
+		document, h.L,
+		"allowDiskUse", "bypassDocumentValidation", "readConcern", "hint", "comment", "writeConcern",
+	)
+
+	var db string
+
+	if db, err = common.GetRequiredParam[string](document, "$db"); err != nil {
+		return nil, err
+	}
+
+	collectionParam, err := document.Get(document.Command())
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO handle collection-agnostic pipelines ({aggregate: 1})
+	// https://github.com/FerretDB/FerretDB/issues/1890
```

Comment:
```suggestion
	// handle collection-agnostic pipelines ({aggregate: 1})
	// TODO https://github.com/FerretDB/FerretDB/issues/1890
```

Our linter [skips that for now](https://github.com/FerretDB/FerretDB/blob/eccf40de450b0ad883cf4495ad208f391a952d3c/tools/checkcomments/checkcomments.go#L46-L50) because there are too many comments like that, but I think we should not add new onces

Reply:
My eyes couldn't spot them :see_no_evil: